frame/dapps-staking/src/tests.rs

```diff
@@ -1205,3 +1209,418 @@ fn new_era_forcing() {
+        advance_era_and_reward(SKIP_ERA, ERA_REWARD, INITIAL_STAKE);
+        bond_and_stake(claimer, contract, STAKE_AMOUNT);
+        advance_era_and_reward(1, ERA_REWARD, INITIAL_STAKE);
+        bond_and_stake(claimer, contract, STAKE_AMOUNT);
+        advance_era_and_reward(SKIP_ERA, ERA_REWARD, INITIAL_STAKE);
+        let claim_era: EraIndex = DappsStaking::current_era().unwrap_or(Zero::zero());
+        claim(claimer, contract, start_era, claim_era.clone());
+
+        cleared_contract_history(contract, FROM_ERA, claim_era);
+    })
+}
+
+//
+// HELPER FUNCTIONS
+//
+
+// helper fn to make  register() one liner for readability
+fn register(developer: u64, contract: SmartContract<AccountId>) {
+    assert_ok!(DappsStaking::register(
+        Origin::signed(developer),
+        contract.clone()
+    ));
+}
+// helper fn to skip "for_era" eras, but reward each era
+fn advance_era_and_reward(
+    for_era: EraIndex,
+    rewards: BalanceOf<TestRuntime>,
+    staked: BalanceOf<TestRuntime>,
+) {
+    // TODO advance era by incrementing block production, needed for block rewards
+    let current: EraIndex = mock::DappsStaking::current_era().unwrap_or(Zero::zero());
+
+    let era_reward = EraReward { rewards, staked };
+    for era in 0..for_era {
+        <PalletEraRewards<TestRuntime>>::insert(current + era, era_reward.clone());
+    }
+    <CurrentEra<TestRuntime>>::put(&current + for_era);
+}
+
+// helper fn to check updated storage items after claim is called
+fn cleared_contract_history(
+    contract: SmartContract<mock::AccountId>,
+    from_era: EraIndex,
+    to_era: EraIndex,
+) {
+    // check claim pointer moved
+    assert_eq!(
+        mock::DappsStaking::contract_last_claimed(&contract).unwrap_or(Zero::zero()),
+        to_era
+    );
+
+    // check last staked pointer moved
+    assert_eq!(
+        mock::DappsStaking::contract_last_staked(&contract).unwrap_or(Zero::zero()),
+        to_era
+    );
+
+    // check new contractEraStaked
+    assert_ok!((mock::DappsStaking::contract_era_stake(contract, to_era))
+        .is_some()
+        .then(|| ())
+        .ok_or("contract_era_stake not created"));
+
+    // check history storage is cleared
+    for era in from_era..to_era {
+        assert_ok!((mock::DappsStaking::contract_era_stake(contract, era))
+            .is_none()
+            .then(|| ())
+            .ok_or("contract_era_stake not cleared"));
+    }
+}
+
+// helper fn to make claim() one liner for readability
+fn claim(
+    claimer: AccountId,
+    contract: SmartContract<mock::AccountId>,
+    start_era: EraIndex,
+    claim_era: EraIndex,
+) {
+    assert_ok!(DappsStaking::claim(Origin::signed(claimer), contract));
+    // check the event for claim
+    System::assert_last_event(mock::Event::DappsStaking(crate::Event::ContractClaimed(
+        contract, claimer, start_era, claim_era,
+    )));
+}
+
+// helper fn to make bond_and_stake() one liner for readability
+fn bond_and_stake(
```

Comment:
There's similar method in testing_utils which you should use.

That one doesn't assert the last event since it's possible that the amount bonded&staked will be less than what was given in the method.